### DISCUSSION TITLE
Merge queue and required status checks

### DISCUSSION THREAD

**Discussion by SaraRavasio** [community/community#39933] [Pull Requests]
### Select Topic Area

Bug

### Body

I have set up merge queue for one of my repositories.
In this repository, I have some required status checks that need to pass in order to be able to merge.

Required status:
- mergefreeze
- lintin
- test coverage

In merge queue:
- cypress tests

The required gh actions pass before merging, but when I start the merge queue and my cypress tests pass, I still see the merge queued due to the requirements for these 3 actions to pass, but they're not even triggered.

<img width="1252" alt="image" src="https://user-images.githubusercontent.com/33936661/203779883-0cac2ea6-fc4d-43cc-bf16-5f581f5eccca.png">

**Comment by ossanna16** [comment]
Hi there @SaraRavasio and welcome to our community! Thank you for reporting this bug!

**Comment by mrmckeb** [comment]
Hi there, you need to use the `merge_group` event:
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#merge_group

We hit this too, it's a strange one - but I think the idea is that they want you to be able to know when something is running in a `pull_request` vs `merge_group`.

**Reply by SaraRavasio** [reply]
Yeah I actually tried that yesterday morning in my ci action and it works as expected 👍 

One thing that I was surprised about it: I thought that failing cypress tests at the merge queue step will prevent my PR from being merged, but that was not the case.

So I had to go with a different approach 😕 not sure if I was doing something wrong.

**Reply by mrmckeb** [reply]
They should prevent merge if they're required checks. Perhaps something else went wrong?

**Reply by SaraRavasio** [reply]
Not sure, what I tried to do was have cypress only in merge queue.
But as I want to run it only when I merge, and not before my PR gets merged, I don't have it as a required status check for my branch.
Maybe that's the reason?

**Reply by mrmckeb** [reply]
I haven't tried that yet, but I think you could pull it off with branch protection.

**Reply by SaraRavasio** [reply]
My setup now is: cypress is a required status check, but it gets triggered only when the PR has a label `run-cypress`

so my team can enable it only when they really need it, for example before merging.

it's running in this condition
```
on:
  pull_request:
    types: [labeled, synchronize]

jobs:
   myjob:
      if: contains(github.event.pull_request.labels.*.name, 'run-cypress')
```

**Reply by rabbiveesh** [reply]
I'm working on the same issue. What I ended up doing is making it a required check for all PRs, and enabling auto-merge, and then starting to run it on the auto_merge_enabled event.

It's suboptimal, b/c it won't get rerun later, so you have to disable + then re-enable auto-merge (which is the same thing you have w/ your label approach).

I would have imagined that anything that runs a test in the merge queue would automatically be considered a required check. Seems like a bug to me

**Reply by SaraRavasio** [reply]
oh i see!
well if it could help, with my code you don't need to remove/add the label. as soon as the label is present, on each commit it will run the action :) maybe it could come in handy!

**Reply by rabbiveesh** [reply]
oh; i didn't read carefully enough. You're right, if I add `synchronize` and an `if` check like you have, then it would run on every push.

Thanks!

**Reply by SaraRavasio** [reply]
@kiajfaty yes, I tried to use `merge_group`, but without success :/

**Reply by rabbiveesh** [reply]
@SaraRavasio new problem; when a job is skipped, it counts as sucess, so you have nothing forcing you to actually run the cypress test in the PR

**Reply by SaraRavasio** [reply]
@rabbiveesh good point, another very annoying (in my opinion) bug on their side.

This is how I overcome this problem:
- in my cypress dashboard I've added a board connected to my repository
<img width="1028" alt="image" src="https://user-images.githubusercontent.com/33936661/206245785-ae682503-46a0-4805-bad2-2efcf49177ad.png">

- if you do that and run again cypress, you'll see that one of the required status check that you have available now is `cypress: default-group`, use that one
<img width="432" alt="image" src="https://user-images.githubusercontent.com/33936661/206245870-ac9e956e-7b79-45fe-8336-469b9d4a4163.png">

- it will always stay in "loading" state until you run your tests, and this means you can't merge your pr
<img width="575" alt="image" src="https://user-images.githubusercontent.com/33936661/206246107-a1f70f2b-b43c-4cd0-9d06-738fc357cbef.png">


Let me know if it works for you :)

**Reply by rabbiveesh** [reply]
ah, that works for cypress; i'm doing a couple of other things.

In the end, i rewrote my pre-master tests to be a reusable workflow, so they won't get called at all if the workflow is in the wrong state, thus never getting skipped, b/c the skip just keeps them from getting called.

I also don't have the requirement to be up to date w/ master, b/c I run the tests again in the merge queue, so even if in theory there's a race condition where the PR gets auto-merged before the workflow call hits (don't know if that happens), it'll end up getting tested on master.

Obviously, that's not an option for cypress (my cypress test suite takes 10 minutes :cry: ), but it's a useful thing nonetheless
